### ISSUE TITLE
Implement Scenes & References

### ISSUE THREAD

**Issue by kevinh145** [moethu/SketchUpSharp#19]
SketchUp makes use of scenes to hide and show a specified set of layers. It would be nice if there was a reference to these scenes.

**Comment by moethu** [comment]
Hi @promontis yes the library doesn't include scenes at the moment. It looks like it would be easy to add them. Do you need them for a specific use-case?

**Comment by promontis** [comment]
@moethu yeah, I will need to read and write scenes. I'm creating a 3d configurator, and the different configurations are expressed as scenes.

**Comment by moethu** [comment]
what parts of a scene would be necessary for you to achieve this? rendering options? layer settings? element visibility?

**Comment by promontis** [comment]
@moethu element visibility... "hidden objects" in the SketchUp UI

**Comment by moethu** [comment]
@promontis I started implementing a reading scenes with hidden elements feature first.
How would you like to access hidden elements? My first shot was adding a list of IDs of hidden elements - but then you'd need a central container for entities in the data model.

**Comment by promontis** [comment]
What options do we have? Elements are now split up in different cases, right? Like components, curves, groups, instances, etc. What about adding a list of ID with the type of element? Does that work?

**Comment by moethu** [comment]
In order to keep compatibility I wouldn't remove the existing lists by types. I added a new dictionary (int ID, entity) so you have both options in the end. objects are only referenced so this doesn't require much additional memory. Each Scene object has a list of hiddenElements containing references to the actual elements. Would this work for you?
So what about writing hidden elements, just allow adding elements to the list of hidden elements?

**Comment by promontis** [comment]
Yeah, both suggestion work for me!

**Comment by moethu** [comment]
Got an issue with writing hidden entities, the API doesn't provide a method for it. There is no `SUSceneSetHiddenEntities`. I'll check if there is any workaround but I implemented a way to get Scenes and their hidden entities.

**Comment by moethu** [comment]
@promontis can you give this branch a try? it works for me. writing scenes however will require some more changes in the library.

**Comment by moethu** [comment]
@promontis I never got your feedback on this one, does it work?
